### DISCUSSION TITLE
CARDINAL wrapper installation.

### DISCUSSION THREAD

**Discussion by AnthonyB08** [idaholab/moose#23408] [Q&A General] [answered]
Hi all, 

I am trying to install CARDINAL wrapper for using MOOSE and NEKS. When I run the command 
./contrib/moose/scripts/update_and_rebuild_petsc.sh
I receive this error:

![Screenshot 2023-02-13 135244](https://user-images.githubusercontent.com/55299169/218548326-04655418-d6d3-42ea-8fc5-6f9459766415.png)

What does the community recommend?

**Comment by milljm** [comment, marked as answer]
Can you attach the configure.log file? It should exist at:

```pre
/mnt/d/cardinal/contrib/moose/petsc/configure.log
```

Also, in the future it would help if you refrained from posting images of text. The issue doing so, is that the search functionality on "Discussions" is unable to parse the information in an image. Thanks!

**Reply by AnthonyB08** [reply]
Okay, I will do that from now on thank you for education me on that issue. Thank you for taking a look at this issue, here is the co
[configure_log.txt](https://github.com/idaholab/moose/files/10726935/configure_log.txt)

**Reply by AnthonyB08** [reply]
@milljm , I managed to get past this part of the CARDINAL installation last week. However, I ran out of storage during the installation and had to uninstall and re-install WSL/Ubuntu onto a different drive. I am wondering if I am missing and ubuntu packages that might correct this error..

**Reply by milljm** [reply]
What version of Ubuntu are you using?

**Reply by milljm** [reply]
I am looking through your log, and I notice you are using perhaps some of our environment (The moose-mpich conda environment perhaps?)

Are you following these instructions: https://cardinal.cels.anl.gov/with_conda.html
(you're only after OpenMC, and therefore you should not be trying to build PETSc)

Or are you instead needing to build without using Conda: https://cardinal.cels.anl.gov/without_conda.html
(you need to install all the compilers yourself using your systems package management)

**Reply by milljm** [reply]
I was able build PETSc, running the `moose/scripts/update_and_rebuild_petsc.sh`, after performing the following apt-get commands on *fresh install of* Ubuntu 22.04. These are similar commands we use to initiate a fresh machine for use within our build farm.

My advice, is to run these only in a new WSL container... I have no idea what havoc these commands might do in a WSL environment you've already established.

```bash
    apt-get update
    apt-get -y --allow-unauthenticated upgrade
    apt-get -y --allow-unauthenticated install software-properties-common
    apt-get -y --allow-unauthenticated install build-essential debootstrap vim emacs gfortran libperl-dev python3-dev python3 file flex bison autoconf automake curl libtirpc-dev git procps unzip python3-yaml gcc-11 g++-11 gfortran-11 cpp-11 libx11-dev libxt-dev libglpk-dev rsync
    apt-get -y remove gcc-12 g++-12 gfortran-12 cpp-12
    apt-get -y autoremove
    apt-get -y clean
    apt-get -y autoclean
    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 100
    update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100
    update-alternatives --install /usr/bin/gfortran gfortran /usr/bin/gfortran-11 100
    update-alternatives --install /usr/bin/cpp cpp /usr/bin/cpp-11 100

    # Installs OpenMPI, Cmake
    apt-get install libopenmpi-dev cmake python3-packaging pkg-config

    # Makes `python` default to `python3` (probably only valid on Ubuntu 22.04!)
    update-alternatives --install /usr/bin/python python /usr/bin/python3.10 2
    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 3

    # Set compilers (you'll need to do this every time you launch your WSL Ubuntu container)
    export CC=mpicc CXX=mpicxx FC=mpif90 F90=mpif90 F77=mpif77
```

**Reply by AnthonyB08** [reply]
This solved all of my issues! I cannot thank you enough! I am excited to start my research now!

**Reply by AnthonyB08** [reply]
I was missing a few of those packages. I am now able to compile and I successfully ran OpenMC through cardinal

**Reply by milljm** [reply]
Your very welcome! I think you might like this upcoming process: https://github.com/neams-th-coe/cardinal/issues/594

It was your discussion here that prompted me to "help out" and make some easier Conda available packages for installation =D

I'll post back here when ready. The idea is you'll only need to:

```bash
mamba create -n cardinal cardinal
```
To have OpenMC, MOAB, dagmc, NekRS, and Cardinal installed.

**Reply by AnthonyB08** [reply]
Oooo that will be exciting and very simple. The reason I didn't go through the conda environment is only that I wanted the capability of NEKRS. Thank you for tackling this problem!
